src/taipy/core/data/parquet.py

```diff
@@ -0,0 +1,222 @@
+        config_id: str,
+        scope: Scope,
+        id: Optional[DataNodeId] = None,
+        name: Optional[str] = None,
+        owner_id: Optional[str] = None,
+        parent_ids: Optional[Set[str]] = None,
+        last_edit_date: Optional[datetime] = None,
+        job_ids: List[JobId] = None,
+        cacheable: bool = False,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        properties: Dict = None,
+    ):
+        if properties is None:
+            properties = {}
+        if missing := set(self._REQUIRED_PROPERTIES) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties " f"{', '.join(x for x in missing)} were not informed and are required"
+            )
+
+        if self.__ENGINE_PROPERTY not in properties.keys():
+            properties[self.__ENGINE_PROPERTY] = "pyarrow"
+
+        if self.__COMPRESSION_PROPERTY not in properties.keys():
+            properties[self.__COMPRESSION_PROPERTY] = "snappy"
+
+        if self.__READ_KWARGS_PROPERTY not in properties.keys():
+            properties[self.__READ_KWARGS_PROPERTY] = {}
+
+        if self.__WRITE_KWARGS_PROPERTY not in properties.keys():
+            properties[self.__WRITE_KWARGS_PROPERTY] = {}
+
+        self._path = properties.get(self.__PATH_KEY, properties.get(self.__DEFAULT_PATH_KEY))
+        if self._path is None:
+            raise MissingRequiredProperty("default_path is required in a Parquet data node config")
+        else:
+            properties[self.__PATH_KEY] = self._path
+
+        if self.__EXPOSED_TYPE_PROPERTY not in properties.keys():
+            properties[self.__EXPOSED_TYPE_PROPERTY] = self.__EXPOSED_TYPE_PANDAS
+        self._check_exposed_type(properties[self.__EXPOSED_TYPE_PROPERTY])
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            name,
+            owner_id,
+            parent_ids,
+            last_edit_date,
+            job_ids,
+            cacheable,
+            validity_period,
+            edit_in_progress,
+            **properties,
+        )
+        if not self._last_edit_date and isfile(self._path):
+            self.unlock_edit()
+
+    @classmethod
+    def storage_type(cls) -> str:
+        return cls.__STORAGE_TYPE
+
+    @property  # type: ignore
+    @_self_reload(DataNode._MANAGER_NAME)
+    def path(self):
+        return self._path
+
+    @path.setter
+    def path(self, value):
+        self._path = value
+        self.properties[self.__PATH_KEY] = value
+
+    def _check_exposed_type(self, exposed_type):
+        if isinstance(exposed_type, str) and exposed_type not in self.__VALID_STRING_EXPOSED_TYPES:
+            raise InvalidExposedType(
+                f"Invalid string exposed type {exposed_type}. Supported values are {', '.join(self.__VALID_STRING_EXPOSED_TYPES)}"
+            )
+
+    def _read(self):
+        return self.read_with_kwargs()
```

Comment:
```suggestion
    def _read(self):
        read_kwargs = properties[self.__READ_KWARGS_PROPERTY]
        return self.read_with_kwargs(read_kwargs)
```

Reply:
I think `read_with_kwargs` needs to be the one that gets the config kwargs because it can be manually called at runtime. Even if we loaded the config kwargs in `_read`, we'd still need to load it in `read_with_kwargs` to address the manual case.

Reply:
Yes but `DataNode.read` calls `ParquetDataNode._read` which calls `ParquetDataNode.read_with_kwargs` (which loads the config kwargs). The config kwargs are used in all exposed read methods (which I believe is the behaviour we want). The `ParquetDataNode.read_with_kwargs` method (compared to `DataNode.read`) only brings the added behaviour that new kwargs may be specified at runtime (such as for column names or filters)

Reply:
Yes OK. I understand!